app/controllers/bookmarks_controller.rb

```diff
@@ -9,7 +9,9 @@ def index
 
   def new
     @bookmark            = Bookmark.new
-    @bookmark.name       = params[:query].to_s.strip.split(/\s| = |!|~|>|</)[0]
+    query                = params[:query].to_s.squish
```

Comment:
auto completer (complete_for) with mouse clicks and keyboard select
(up/down etc) many times add excessive whitespaces, squish is equal to
strip and ensure a whitespace exists only once. imho this looks better when
saving the search string.